bb8/src/api.rs

```diff
@@ -406,8 +407,11 @@ pub trait CustomizeConnection<C: Send + 'static, E: 'static>:
     ///
     /// The default implementation simply returns `Ok(())`. If this method returns an
     /// error, it will be forwarded to the configured error sink.
-    async fn on_acquire(&self, _connection: &mut C) -> Result<(), E> {
-        Ok(())
+    fn on_acquire<'r, 'a: 'r, 'b: 'r>(
```

Comment:
Looks like we can actually make do with a single lifetime, so please change it to just have `'a` for both arguments and the resulting `dyn Future`.